src/en/RoyalRoad.lua

```diff
@@ -142,9 +319,29 @@ return {
 	end,
 
 	search = function(data)
-		return parseListing(GETDocument(qs({
-			title = data[QUERY]
-		}, baseURL .. "/fictions/search")))
+		return parseListing(GETDocument(baseURL .. "/fictions/search" .. createFilterString(data)))
 	end,
-	isSearchIncrementing = false
+	isSearchIncrementing = false,
+	searchFilters = {
+		TextFilter(KEYWORD_FILTER_KEY, "Keyword (title or description)"),
+		TextFilter(AUTHOR_FILTER_KEY, "Author name"),
+		FilterGroup("Genres", MultiTriStateFilter(200, GENRES_FILTER_EXT, 15)),
+		FilterGroup("Additional Tags", MultiTriStateFilter(300, TAGS_FILTER_EXT, 46)),
+		FilterGroup("Content Warnings", MultiTriStateFilter(400, CONTENT_WARNINGS_FILTER_EXT, 4)),
```

Comment:
Avoid repeated use of magic constants.

```suggestion
		FilterGroup("Genres", MultiTriStateFilter(GENRES_FILTER_KEY, GENRES_FILTER_EXT, #GENRES_FILTER_EXT)),
		FilterGroup("Additional Tags", MultiTriStateFilter(TAGS_FILTER_KEY, TAGS_FILTER_EXT, #TAGS_FILTER_EXT)),
		FilterGroup("Content Warnings", MultiTriStateFilter(CONTENT_WARNINGS_FILTER_KEY, CONTENT_WARNINGS_FILTER_EXT, #CONTENT_WARNINGS_FILTER_EXT)),
```